src/number_theory/cyclotomic/basic.lean

```diff
@@ -204,6 +209,104 @@ lemma number_field [h : number_field K] [fintype S] [is_cyclotomic_extension S K
     (@algebra_rat L _ (char_zero_of_injective_algebra_map (algebra_map K L).injective)) _ _
     h.to_finite_dimensional (finite S K L) }
 
+/-- A finite cyclotomic extension of an integral noetherian domain is integral -/
+lemma integral [is_domain B] [is_noetherian_ring A] [fintype S] [is_cyclotomic_extension S A B] :
+  algebra.is_integral A B :=
+is_integral_of_noetherian $ is_noetherian_of_fg_of_noetherian' $ (finite S A B).out
+
+/-- If `S` is finite and `is_cyclotomic_extension S K A`, then `finite_dimensional K A`. -/
+lemma finite_dimensional [fintype S] [algebra K A] [is_domain A] [is_cyclotomic_extension S K A] :
+  finite_dimensional K A :=
+finite S K A
+
 end fintype
 
+section
+
+variables {A B}
+
+lemma adjoin_roots_cyclotomic_eq_adjoin_nth_roots [decidable_eq B] [is_domain B] {ζ : B}
+  (hζ : is_primitive_root ζ n) :
+  adjoin A ↑((map (algebra_map A B) (cyclotomic n A)).roots.to_finset) =
+  adjoin A {b : B | ∃ (a : ℕ+), a ∈ ({n} : set ℕ+) ∧ b ^ (a : ℕ) = 1} :=
+begin
+  simp only [mem_singleton_iff, exists_eq_left, map_cyclotomic],
+  refine le_antisymm (adjoin_mono (λ x hx, _)) (adjoin_le (λ x hx, _)),
+  { simp only [multiset.mem_to_finset, finset.mem_coe,
+               map_cyclotomic, mem_roots (cyclotomic_ne_zero n B)] at hx,
+    simp only [mem_singleton_iff, exists_eq_left, mem_set_of_eq],
+    rw is_root_of_unity_iff n.pos,
+    exact ⟨n, nat.mem_divisors_self n n.ne_zero, hx⟩ },
+  { simp only [mem_singleton_iff, exists_eq_left, mem_set_of_eq] at hx,
+    obtain ⟨i, hin, rfl⟩ := hζ.eq_pow_of_pow_eq_one hx n.pos,
+    refine set_like.mem_coe.2 (subalgebra.pow_mem _ (subset_adjoin _) _),
+    rwa [finset.mem_coe, multiset.mem_to_finset, mem_roots $ cyclotomic_ne_zero n B],
+    exact is_root_cyclotomic n.pos hζ }
+end
+
+end
+
+section field
+
+variable [ne_zero ((n : ℕ) : K)]
+
+/-- If `ne_zero ((n : ℕ) : K)`, a cyclotomic extension splits `X ^ n - 1` if `n ∈ S`.-/
+lemma splits_X_pow_sub_one [H : is_cyclotomic_extension S K L] (hS : n ∈ S) :
```

Comment:
If `cyclotomic n K` splits, then `X ^ n - 1` also splits, right? Should this be a general lemma, so that this current lemma is a one-line consequence of the next one?

Reply:
The implication goes the other way, `cyclotomic n K` divides `X ^ n - 1` and so it splits :) (this is indeed the proof).